src/common.ts

```diff
@@ -0,0 +1,127 @@
+import filenamify from 'filenamify';
+import fs from 'fs-extra';
+import os from 'os';
+import path from 'path';
+import createDebug from 'debug';
+import { ComboOptions, Options } from './types';
+import { CreateOptions as AsarOptions } from '@electron/asar';
+
+export const debug = createDebug('electron-packager');
+
+export function sanitizeAppName(name: string) {
+  return filenamify(name, { replacement: '-' });
+}
+
+export function generateFinalBasename(opts: Pick<ComboOptions, 'arch' | 'name' | 'platform'>) {
+  return `${sanitizeAppName(opts.name!)}-${opts.platform}-${opts.arch}`;
+}
+
+export function generateFinalPath(opts: ComboOptions) {
+  return path.join(opts.out || process.cwd(), generateFinalBasename(opts));
+}
+
+export function info(message: unknown, quiet?: boolean) {
+  if (!quiet) {
+    console.info(message);
+  }
+}
+
+export function warning(message: unknown, quiet?: boolean) {
+  if (!quiet) {
+    console.warn(`WARNING: ${message}`);
+  }
+}
+
+export function subOptionWarning(properties: Record<string, unknown>, optionName: string, parameter: string, value: unknown, quiet?: boolean) {
+  if (Object.prototype.hasOwnProperty.call(properties, parameter)) {
+    warning(`${optionName}.${parameter} will be inferred from the main options`, quiet);
+  }
+  properties[parameter] = value;
+}
+
+export function createAsarOpts(opts: ComboOptions): false | AsarOptions {
+  let asarOptions;
+  if (opts.asar === true) {
+    asarOptions = {};
+  } else if (typeof opts.asar === 'object') {
+    asarOptions = opts.asar;
+  } else if (opts.asar === false || opts.asar === undefined) {
+    return false;
+  } else {
+    warning(`asar parameter set to an invalid value (${opts.asar}), ignoring and disabling asar`, opts.quiet);
+    return false;
+  }
+
+  return asarOptions;
+}
+
+export function ensureArray<T>(value: T | T[]): T[] {
+  return Array.isArray(value) ? value : [value];
+}
+
+export function isPlatformMac(platform: ComboOptions['platform']) {
+  return platform === 'darwin' || platform === 'mas';
+}
+
+// @TODO(erikian): the only uses of this function are in its own test - remove it?
```

Comment:
I left it there because it might be useful in the future. I don't have a strong opinion on whether to keep it currently.

Reply:
Let's remove it, it hasn't been used for a good 6 years and we can always recover it from the Git history if necessary - https://github.com/electron/packager/pull/1593/commits/32af02c78e2f62a741b49b30502b3acbd38fbb96